### DISCUSSION TITLE
⌘9 behavior on macOS

### DISCUSSION THREAD

**Discussion by Samasaur1** [ghostty-org/ghostty#3361] [Feature Requests, Ideas] [answered]
This is a combination of a question about changing the default behavior and something that is kind of bug-like.

#### Behavior

In Terminal.app and every other Mac app I can think of, ⌘9 goes to the last tab instead of the 9th tab when more than 9 tabs exist. In Ghostty, however, ⌘9 selects the 9th tab in that scenario.

#### Bug

I was able to override this behavior with
```
keybind = command+nine=last_tab
```
but this did not change the little hints that Ghostty inserts for each tab:
<img width="1512" alt="screenshot of the tab bar with 10 tabs showing" src="https://github.com/user-attachments/assets/5eedcb95-c720-41cd-9642-af0cf5a3ede8" />

#### Next steps

I'm not really sure what exactly I'm asking for here. Mostly I guess I'm saying that I'm surprised that ⌘9 *didn't* go to the last tab with no configuration, since that's the "usual" Mac app behavior. It would also be nice if the hints in the tab bar updated if you set keybindings like I have done, although I imagine that the current hints are hardcoded and getting this to work would be a not insignificant amount of work for very little benefit.

**Comment by gpanders** [comment]
> It would also be nice if the hints in the tab bar updated if you set keybindings

It certainly *used* to do this. I wonder if this was changed unintentionally? I’m not at my computer right now, but can take a look later to double check.

**Reply by gpanders** [reply]
I verified that the tab bar labels _do_ still update based on their keybindings. Here I've added `keybind = option+2=goto_tab:2` to my config:

<img width="912" alt="Screenshot 2024-12-27 at 9 47 36 AM" src="https://github.com/user-attachments/assets/448b1470-fd92-451a-81d8-9a5c4b765b32" />

**Reply by gpanders** [reply]
>In Terminal.app and every other Mac app I can think of, ⌘9 goes to the last tab instead of the 9th tab when more than 9 tabs exist.

I actually didn't know this, but it does seem to be the case in the applications I've tried.

**Comment by qwerasd205** [comment]
This would simply be a matter of updating the default binding. Edit: oop, I didn't read the full post. Yeah I agree that it should be like that by default.

**Comment by mitchellh** [comment, marked as answer]
#3528

**Comment by lilyball** [comment]
Dash actually has different behavior here, which is that ⌘9 goes to the 9th tab, unless you're on the 9th tab, then it goes to the last tab. I didn't actually realize that Safari and Terminal.app don't do this since I don't normally keep more than 9 tabs in those windows. I really like Dash's behavior here since it gives you the option of which you want, although were I to design this feature in a vacuum I might flip the order of those (having ⌘9 go to the last tab, unless it's already on the last tab then it would go to the 9th tab).

**Reply by mitchellh** [reply]
I think this feature is somewhat esoteric enough that saying "hey, we just match Terminal.app and Safari" is more reasonable here. I don't think we need to optimize this UX a ton otherwise we have to defend our opinions more rigorously there.
